src/App.tsx

```diff
@@ -1,25 +1,46 @@
-import React from 'react';
-import './App.css';
+import React, { useState } from 'react';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
+import './App.css';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(5);
```

Comment:
It seems that 'perPage' state is never updated after the initial render, except in 'handlePerPageChange'. If 'perPage' should not change during the component's lifecycle, consider using a constant instead of a state variable. If it should be changeable, ensure there is a way for it to be updated accordingly.